dbt_semantic_interfaces/implementations/metric.py

```diff
@@ -228,3 +238,31 @@ def input_metrics(self) -> Sequence[PydanticMetricInput]:
             return (self.type_params.numerator, self.type_params.denominator)
         else:
             assert_values_exhausted(self.type)
+
+    @staticmethod
+    def all_input_measures_for_metric(
+        metric: Metric, metric_index: Dict[MetricReference, Metric]
+    ) -> Set[MetricInputMeasure]:
+        """Gets all input measures for the metric, including those defined on input metrics (recursively)."""
+        measures = set()
+        if metric.type is MetricType.SIMPLE or metric.type is MetricType.CUMULATIVE:
+            assert (
+                metric.type_params.measure is not None
+            ), f"Metric {metric.name} should have a measure defined, but it does not."
+            measures.add(metric.type_params.measure)
```

Comment:
Not blocking, but heads up that this could end up with duplicate measures.

Ie.,
if a metric that has 2 measures is the same but has a different input filter, then the `set()` wouldn't count them as duplicates as the `MetricInputMeasure` will have a different `filter` field even tho it's referencing the same measure. 

If you don't want duplicate measures showing up in this, you could return `Set[MeasureReference]` instead.

Reply:
Good call!!